### PR TITLE
Feature insert tooltips for icons

### PR DESCRIPTION
**Pull request by melanieretter**
* Insert tooltips for all icons in the UI where necessary.
* Insert class that holds the constants of the tooltip message keys.
* The caption of the Action History Table on the Deployment View is shown even when no target is selected.

Signed-off-by: Melanie Retter <melanie.retter@bosch-si.com>

### CODE REVIEW

**Comment by schabdo** [comment]
@hawkbit-bot verify please

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/ConfigMenuBar.java [hunk 1]
```diff
@@ -66,14 +70,15 @@
      */
     public ConfigMenuBar(final boolean createPermission, final boolean updatePermission, final boolean deletePermission,
```
[the remark this answers is not in the document]
Comment (melanieretter) [reply hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/components/ConfigMenuBar.java]: done

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/UIMessageIdProvider.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    public static final String TOOLTIP_ROLLOUT_PAUSE = "tooltip.rollout.pause";
+
+    public static final String TOOLTIP_ROLLOUT_RUN = "tooltip.rollout.run";
+
+    public static final String TOOLTIP_ROLLOUT_APPROVE = "tooltip.rollout.approve";
+
+    public static final String TOOLTIP_IN_TIME = "tooltip.in.time";
+
+    /**
+     * /* Private Constructor.
```
Comment (laverman) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/utils/UIMessageIdProvider.java]: commentary

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionHistoryLayout.java [hunk 3]
```diff
@@ -58,9 +60,16 @@ public ActionHistoryLayout(final VaadinMessageSource i18n, final DeploymentManag
         this.deploymentManagement = deploymentManagement;
         this.notification = notification;
         this.managementUIState = managementUIState;
+        actionHistoryCaption = getActionHistoryCaption("");
         init();
     }
 
+    private String getActionHistoryCaption(final String targetName) {
+        return HawkbitCommonUtil.DIV_DESCRIPTION_START
```
Comment (laverman) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionHistoryLayout.java]: Default should be "Action History" and after a target is selected "Action History for ..."

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGridComponentLayout.java [hunk 4]
```diff
@@ -28,7 +28,7 @@
     private static final long serialVersionUID = -3766179797384539821L;
 
     protected final transient EventBus.UIEventBus eventBus;
-    protected final VaadinMessageSource i18n;
+    protected VaadinMessageSource i18n;
```
Comment (schabdo) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/grid/AbstractGridComponentLayout.java]: If you introduce a getter, couldn't this be private?

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionHistoryLayout.java [hunk 5]
```diff
@@ -58,17 +61,31 @@ public ActionHistoryLayout(final VaadinMessageSource i18n, final DeploymentManag
...
+        actionHistoryCaption = getActionHistoryCaption();
         init();
     }
 
+    private final String getActionHistoryCaption() {
+        return getActionHistoryCaption(null);
+    }
+
+    private String getActionHistoryCaption(final String targetName) {
+        String caption = getI18n().getMessage("caption.action.history");
```
Comment (stefbehl) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/management/actionhistory/ActionHistoryLayout.java]: "for" should not be hardcoded... I'd recomment to define two messages with two separate keys...

**Pull request merged**